lib/Db/MessageMapper.php

```diff
@@ -235,7 +235,7 @@ public function writeThreadIds(array $messages): void {
 			foreach ($messages as $message) {
 				$query->setParameter(
 					'thread_root_id',
-					$message->getThreadRootId(),
+					mb_substr($message->getThreadRootId(), 0, 1023),
```

Comment:
How often will the prefix of `thread_root_id` be the same causing collisions? Would hashing to 1k bytes provide better resiliency against collisions?

Reply:
Hard to say. Hashing long values could be a good idea for long messages. I'll look into that!